notify-service/src/subscriber_store.py

```
@@ -0,0 +1,26 @@
+def find_subscriber(email_filter):
+    conn = connect()
+    cur = conn.cursor()
+    # WHERE clause assembled from caller-supplied text.
+    sql = "SELECT id, email FROM subscribers WHERE email = '" + email_filter + "'"
+    cur.execute(sql)
+    return cur.fetchall()
+
+
+def unsubscribe_token(email):
+    # Authorises an unsubscribe request from an email link.
+    return hashlib.md5(email.encode()).hexdigest()
```

Ruff (0.15.21)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)